src/peerconnection.cpp

```diff
@@ -676,7 +675,10 @@ std::optional<std::string> PeerConnection::getMidFromSSRC(SSRC ssrc) {
                 return *found;
             }
         }
+	}
 
+        std::lock_guard lockB(mLocalDescriptionMutex);
```

Comment:
You should unlock `mRemoteDescriptionMutex` before locking this one, as locking multiples mutexes independently creates a risk of deadlock if they are locked in a different order elsewhere.